src/prefect/blocks/core.py

```diff
@@ -690,37 +690,59 @@ async def register_type_and_schema(cls, client: "OrionClient" = None):
                     if Block.is_block_class(type_):
                         await type_.register_type_and_schema(client=client)
 
-        exception_stack = list()
         try:
             block_type = await client.read_block_type_by_slug(
                 slug=cls.get_block_type_slug()
             )
+            cls._block_type_id = block_type.id
             await client.update_block_type(
                 block_type_id=block_type.id, block_type=cls._to_block_type()
             )
         except prefect.exceptions.ObjectNotFound:
             block_type = await client.create_block_type(block_type=cls._to_block_type())
-        except PrefectHTTPStatusError as exc:
-            exception_stack.append(exc)
-
-        cls._block_type_id = block_type.id
+            cls._block_type_id = block_type.id
+        # Block protection for block types update has been removed, but this code
+        # is kept for compatibility with version 2.4.2 servers. If support for
+        # 2.4.2 servers is dropped in the future, this check can safely
+        # be removed.
+        except prefect.exceptions.ProtectedBlockError as exc:
+            if cls._block_type_id is None:
+                raise RuntimeError(
+                    "Unable to register block type due to protected block error. "
+                    "Consider upgrading your Prefect client and/or server to the "
+                    "latest version to resolve this error."
+                ) from exc
```

Comment:
I wanted to remove the logic handling block protection errors within registration, but the compatibility test for 2.4.2 would fail if I did. I think it's important to consider if newer clients are expected to be compatible with older servers. It will be difficult to add new API routes to simplify things like block registration if it needs to continue to work with older servers.

Reply:
New client -> Old server is not strictly supported. 
Old Client -> New server must be supported. 

If you would like to add an exclusion for the 2.4.2 client -> a 2.5.0 server that's fine with me.